WcaOnRails/app/helpers/test_script.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+wrt_team = FactoryBot.create :team
+user_1 = FactoryBot.create :user
+team_member_1 = FactoryBot.create :team_member, user_id: user_1.id, team_id: wrt_team.id, start_date: Time.now-10.days
+user_2 = FactoryBot.create :user
+team_member_2 = FactoryBot.create :team_member, user_id: user_2.id, team_id: wrt_team.id, start_date: Time.now-4.days
+user_3 = FactoryBot.create :user
+team_member_3 = FactoryBot.create :team_member, user_id: user_3.id, team_id: wrt_team.id, start_date: Time.now-3.months, end_date: Time.now-11.days
+user_4 = FactoryBot.create :user
+team_member_4 = FactoryBot.create :team_member, user_id: user_4.id, team_id: wrt_team.id, start_date: Time.now-5.months, end_date: Time.now-16.days
+puts(team_member_1, team_member_2, team_member_3, team_member_4)
+puts('Number of team members:', wrt_team.team_members.count)
+puts('Object of first team member:', wrt_team.team_members.first)
```

Comment:
This code yields an unintuitive output when run with `bin/rails r path/to/test_script.rb`.

The `team_member_*` are all being printed as (apparently) valid `TeamMember` instances and `team_members.count` is printed as `4`. But `team_members.first` yields `nil` and the printed output stays blank.

My guess is that this happens because we're only mocking stuff with `FactoryBot` and these are not real `ApplicationRecord` models, so somewhere something goes wrong in the deep abyss of the Ruby Interpreter. But that's just a vague suspicion. Do you have any ideas @jonatanklosko?

Reply:
Note that `team_members.count` always issues a count query to the db, so it makes sense that the result is accurate.

My first guess would be that the `team_members` association is already loaded on creation and it's an empty list (perhaps it gets loaded in a validation), so `first` is indeed nil. If that's the case then we can call `wrt_team.team_members.reload` after creating the team members, to make sure we have the up-to-date list :)

Reply:
Ah, that makes a lot of sense! I guess I was stuck too much with the "FactoryBot only does mocks" perspective that I didn't realise an actual DB call might be the issue here. Thanks for clearing that up 🙃 

@danieljames-dj Changing the last line to use a `reload` call like Jonatan suggested fixes the issue!

Reply:
> FactoryBot only does mocks

It actually doesn't mock at all, it is just a convenience for creating actual records :)

Reply:
Thank you @jonatanklosko and @gregorbg :-)